compiler-cli/src/dependencies.rs

```diff
@@ -612,9 +613,32 @@ impl ProvidedPackageSource {
     }
 }
 
+impl PartialEq for ProvidedPackageSource {
+    fn eq(&self, other: &ProvidedPackageSource) -> bool {
+        match (self, other) {
+            (
+                ProvidedPackageSource::Local { path: own_path },
+                ProvidedPackageSource::Local { path: other_path },
+            ) => is_same_file(own_path, other_path).unwrap_or(false),
+            (
+                ProvidedPackageSource::Git {
+                    repo: own_repo,
+                    commit: own_commit,
+                },
+                ProvidedPackageSource::Git {
+                    repo: other_repo,
+                    commit: other_commit,
+                },
+            ) => own_repo == other_repo && own_commit == other_commit,
+            _ => false,
```

Comment:
Can we list al the cases rather than having a catch all here? We've had multiple bugs from variants being added but matches like these silently returning the wrong value rather than having to be modified for the new variant.